models/turbine_models/custom_models/sd_inference/tokenization.py

```diff
@@ -4,415 +4,171 @@
 
-
-def get_tokenized_inputs(
-    pipe,
-    tokenizer,
-    prompt,
-    uncond_prompt,
-    max_length,
-    max_embeddings_multiples: Optional[int] = 8,
-    no_boseos_middle: Optional[bool] = True,
-    skip_parsing: Optional[bool] = False,
-    skip_weighting: Optional[bool] = False,
-):
-    if not skip_parsing:
-        prompt_tokens, prompt_weights = get_prompts_with_weights(
-            tokenizer, prompt, max_length - 2
+        text_inputs = pipe.tokenizer(
+            prompt,
+            padding="max_length",
+            max_length=pipe.model_max_length,
+            truncation=True,
+            return_tensors="pt",
         )
-        if uncond_prompt is not None:
-            uncond_tokens, uncond_weights = get_prompts_with_weights(
-                tokenizer, uncond_prompt, max_length - 2
+        text_input_ids = text_inputs.input_ids
+        untruncated_ids = pipe.tokenizer(
+            prompt, padding="longest", return_tensors="pt"
+        ).input_ids
+
+        if untruncated_ids.shape[-1] >= text_input_ids.shape[-1] and not torch.equal(
+            text_input_ids, untruncated_ids
+        ):
+            removed_text = pipe.tokenizer.batch_decode(
+                untruncated_ids[:, pipe.model_max_length - 1 : -1]
             )
-    else:
-        prompt_tokens = [
-            token[1:-1]
-            for token in tokenizer(
-                prompt, max_length=max_length, truncation=True
-            ).input_ids
-        ]
-        prompt_weights = [[1.0] * len(token) for token in prompt_tokens]
-        if uncond_prompt is not None:
-            if isinstance(uncond_prompt, str):
-                uncond_prompt = [uncond_prompt]
-            uncond_tokens = [
-                token[1:-1]
-                for token in tokenizer(
-                    uncond_prompt, max_length=max_length, truncation=True
-                ).input_ids
-            ]
-            uncond_weights = [[1.0] * len(token) for token in uncond_tokens]
-
-    # round up the longest length of tokens to a multiple of (model_max_length - 2)
-    max_length = max([len(token) for token in prompt_tokens])
-    if uncond_prompt is not None:
-        max_length = max(max_length, max([len(token) for token in uncond_tokens]))
-    max_embeddings_multiples = min(
-        max_embeddings_multiples,
-        (max_length - 1) // (pipe.model_max_length - 2) + 1,
-    )
-    max_embeddings_multiples = max(1, max_embeddings_multiples)
-
-    max_length = (pipe.model_max_length - 2) * max_embeddings_multiples + 2
-
-    # pad the length of tokens and weights
-    bos = tokenizer.bos_token_id
-    eos = tokenizer.eos_token_id
-    prompt_tokens, prompt_weights = pad_tokens_and_weights(
-        prompt_tokens,
-        prompt_weights,
-        max_length,
-        bos,
-        eos,
-        no_boseos_middle=no_boseos_middle,
-        chunk_length=pipe.model_max_length,
-    )
+            print("The following text was removed due to truncation:", removed_text)
```

Comment:
Did you mean to use `print` like this? Generally not great to spew to stdout from a library. Maybe use `warnings.warn(...)` instead?